transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/TransformationPipelineBuilder.scala

```diff
@@ -180,6 +179,82 @@ object TransformationPipelineBuilder extends PipelineBuilder[Args] {
         .groupByKey
     }
 
+    // Get analysis step objects
+    val analysisStepRuns = readRawEntities(EncodeEntity.AnalysisStepRun)
+    val analysisStepVersionsById = readRawEntities(EncodeEntity.AnalysisStepVersion)
+      .withName("Key analysis step versions by ID")
+      .keyBy(_.read[String]("@id"))
+    val analysisStepsById = readRawEntities(EncodeEntity.AnalysisStep)
+      .withName("Key analysis steps by ID")
+      .keyBy(_.read[String]("@id"))
+    val filesByStepRun = fileInputs
+      .withName("Key files by step run ID")
+      .keyBy(_.tryRead[String]("step_run").getOrElse(""))
```

Comment:
This `getOrElse("")` part _works_, but it feels a little hacky. Is there a better way of doing this?

Reply:
Hrm, two options:
1. `filter` the `fileInputs` first to only those that include a `step_run` field, then use `read` instead of `tryRead`
2. Use `flatMap` instead of `keyBy`, and output an `Option[(String, Msg)]`

2 is what I'd try first. I _think_ scio treats `SCollection[(K, V)]` the same as a collection of `V` keyed by `K`